packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -687,10 +687,16 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
 
     const hasData = data !== undefined
 
+    // error is the last known error we have tracked - or if none has been tracked yet the last errored result for the current args
+    let error = currentState.isError ? currentState.error : lastResult?.error
+    if (error === undefined) error = currentState.error
+
+    const hasError = error !== undefined
+
     // isFetching = true any time a request is in flight
     const isFetching = currentState.isLoading
     // isLoading = true only when loading while no data is present yet (initial load with no data in the cache)
-    const isLoading = !hasData && isFetching
+    const isLoading = !hasError && !hasData && isFetching
```

Comment:
```suggestion
    const isLoading = (!lastResult || lastResult.isLoading || lastResult.isUninitialized) && !hasData && isFetching
```

Does this fix your bug, too?

Reply:
Yes, It does look to pass the test. Will convert to use this.